src/lib/bmp/bmp.c

```diff
@@ -0,0 +1,226 @@
+/**
+ * @file
+ * @brief lib for read and draw BMP
+ *
+ * @date Apr 18, 2019
+ * @author Filipp Chubukov
+ */
+
+#include <stdlib.h>
+#include <unistd.h>
+#include <stdio.h>
+#include <fcntl.h>
+#include <sys/mman.h>
+#include <sys/ioctl.h>
+
+#include <linux/fb.h>
+
+/* now it works with non pallete bmp formats - 24 and 32 bits, tested on 32bpp fb and not tested on 16bpp fb */
+typedef struct {
+	unsigned short bfType;	      /* == "BM" */
+	unsigned int bfSize;	      /* Size of file */
+	unsigned int bfReserved;      /* == "00" */ 
+	unsigned int bfOffBits;	      /* Indicate to start of image bytes */
+	unsigned int biSize;	      /* Size of header */
+	int biWidth;		      /* Width of image in pixels */
+	int biHeight;		      /*Height of image in pixels */
+	unsigned int biPlanes;	      /* == 1 */
+	int biBitCount;		      /* bits per pixel */
+	int biCompression;	      /* type of compression(we are not using) */
+	unsigned int biSizeImage;     /* size of image, == 0 if no compression */
+	unsigned int biXPelsPerMeter; /* X resolution */
+	unsigned int biYPelsPerMeter; /* Y resolution */
+	unsigned int biClrUsed;	      /* Number of used from color table colors */
+	unsigned int biClrImportant;  /* Number of important colors - min for drawing */ 
+} bmp_header;
+
+typedef struct {
+	unsigned char red;
+	unsigned char green;
+	unsigned char blue;
+} color;
+
+void read_pallete(int fd, color pallete[], int colors) {
+	unsigned char tmp[4];
+	int i;
+
+	lseek(fd, 53, SEEK_SET);
```

Comment:
What is `53` constant? Replace it with some meaningful macro

Reply:
replaced